### ISSUE TITLE
GetChildOrNull<T>(int idx) thows exception if index is out of bounds

### ISSUE THREAD

**Issue by markdibarry** [godotengine/godot#57128]
### Godot version

4.0 alpha1

### System information

Windows 10

### Issue description

`GetNodeOrNull<T>(string path)` returns `null` if the path is invalid. `GetChildOrNull<T>(int idx)` instead throws an index out of bounds exception if the index is incorrect. I feel this is an oversight, and should probably behave similarly.

### Steps to reproduce

N/A

**Comment by raulsntos** [comment]
In LINQ there is something similar: `ElementAt` and `ElementAtOrDefault`.

The method [`ElementAtOrDefault`](https://source.dot.net/#System.Linq/System/Linq/ElementAt.cs,0bd04e5c179c3c7d,references) does return `default` (in our case `null` since all nodes are reference types) if the index is out of bounds.

So since C# developers will be familiar with LINQ I think it makes sense that this would be the expected behavior.

---

As a workaround you can create your own extension method that checks the index is not out of bounds first:

```csharp
public static T MyGetChildOrNull<T>(this Node node, int idx)
{
    return idx >= 0 && idx < node.GetChildCount() ? node.GetChildOrNull<T>(idx) : null;
}
```

**Comment by markdibarry** [comment]
Yeah I made an extension as a work around for myself until this can get fixed up. I think it's reasonable to make the LINQ-esque behavior be the default since this method doesn't exist in gdscript anyway, so we're kinda free to.

**Comment by cgbeutler** [comment]
This kinda feels more like a proposal...?

Also, for those seeking workarounds, I feel like the more C# way to write the extension would be a `Try...` method. Nullable returns are steadily becoming a faux pas as C# evolves toward the `nullable` feature flag being standard (Lawd knows its saved me a hundred times by now!):
```csharp
    public static bool TryGetChild<TNode>( this Node node, int index, out TNode child ) where TNode : Node
    {
      child = index >= 0 && index < node.GetChildCount() ? node.GetChildOrNull<TNode>(index) : null!;
      return child != null;
    }
```

**Comment by markdibarry** [comment]
Closing in lieu of a proposal.

**Comment by akien-mga** [comment]
Fixed by #64781.
